hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestHoodieSparkSqlWriter.scala

```diff
@@ -1128,12 +1128,14 @@ class TestHoodieSparkSqlWriter {
       new JavaSparkContext(sc), "", table_path, hoodieFooTableName,
       mapAsJavaMap(compactionOperation)).asInstanceOf[SparkRDDWriteClient[HoodieRecordPayload[Nothing]]]
 
-    client.scheduleCompaction(org.apache.hudi.common.util.Option.empty())
-    client.close()
+    val compactionInstant = client.scheduleCompaction(org.apache.hudi.common.util.Option.empty()).get()
 
+    client.compact(compactionInstant)
```

Comment:
client.compact(compactionInstant) cannot be executed here, or the compression will be completed.

Reply:
@Aiden-Dong Thanks for raising this.  Actually, `client.compact()` executes the compaction by writing the new base files without committing to the timeline of the table, i.e., the compaction instant is still inflight in the timeline.  So this is fine.